pageserver/src/tenant.rs

```diff
@@ -1830,6 +1830,18 @@ impl Tenant {
         ctx: RequestContext,
     ) -> Result<Arc<Timeline>, TimelineArchivalError> {
         info!("unoffloading timeline");
+
+        // We activate the timeline below manually, so this must be called on an active timeline.
+        // We expect callers of this function to ensure this.
+        match self.current_state() {
+            TenantState::Activating { .. }
+            | TenantState::Attaching
+            | TenantState::Broken { .. } => {
+                panic!("Timeline expected to be active")
+            }
```

Comment:
Wondering why we chose to panic instead of propagating the error. Is the expectation that this condition should never fail? Maybe it is also worthwhile to add the expection/pre-condition in doc comments.

Reply:
Yeah, if you have not ensured (via errors) that the tenant is active when calling, then there is a bug in your code. After that, it can only shut down, but not return back to activating or attaching states. So it's fair to panic.